### PR TITLE
Deliverable 1

### PR DESCRIPTION
**Pull request by AlanMalagon**
It regards mini-challenges 1 and 2

### CODE REVIEW

File: src/components/App/App.component.jsx [hunk 1]
```diff
@@ -7,31 +9,35 @@ import LoginPage from '../../pages/Login';
...
-    return () => {
-      clearInterval(intervalId);
-      body.removeEventListener('click', rotateBackground);
-    };
-  }, []);
+  //   return () => {
+  //     clearInterval(intervalId);
+  //     body.removeEventListener('click', rotateBackground);
+  //   };
+  // }, []);
```
Comment (PacoMojica) [inline src/components/App/App.component.jsx:35]: You can delete this code and the files if you don't need it

File: src/components/Header/Searchbar/Searchbar.component.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+`;
+
+const InnerContainer = styled(motion.div)`
+    align-items: center;
+    display: flex;
+    margin: auto;
+    width: 50%;
+`;
+
+const StyledSearchbar = styled(motion.input)`
```
Comment (PacoMojica) [inline src/components/Header/Searchbar/Searchbar.component.jsx:19]: This looks good, you can improve this by moving the styled components to their own files, this improves maintainability and readability

File: src/components/VideosGrid/VideoGridItem/VideoGridItem.component.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    min-height: 280px;
+    text-align: left;
+`;
+
+const VideoThumbnail = styled.img`
+    max-height: 180px;
+    object-fit: cover;
+`;
+
+const VideoInfo = styled.div`
```
Comment (PacoMojica) [inline src/components/VideosGrid/VideoGridItem/VideoGridItem.component.jsx:22]: As I mentioned you can move the styled components to their own files, when you have a lot of styled components it is easier to understand what's going when the code is ordered in a logical way, if there is a file that only contains the styles when someone needs to know something about styles just go to that file, it's not a rule, it depends on your preference or your team's preferences

File: src/components/VideosGrid/VideoGridItem/VideoGridItem.component.jsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+export const parseDatetime = (datetime) => {
+    const original = new Date(datetime);
+    const formatted = new Intl.DateTimeFormat('es', {
+        year: 'numeric',
+        month:'short',
+        day: '2-digit',
+        timeZone: 'UTC'
+    }).format(original);
+    return formatted;
+};
```
Comment (PacoMojica) [inline src/components/VideosGrid/VideoGridItem/VideoGridItem.component.jsx:50]: Great, your custom parser, this is not inherent function of the component so you can have a utils folder for functions like this one

File: src/components/VideosGrid/VideoGridItem/VideoGridItem.test.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        const title = screen.getByText(/mock video title/i);
+        const info = screen.getByText(/mock channel title • 27 sept. 2014/i);
+        expect(title).toBeInTheDocument();
+        expect(info).toBeInTheDocument();
+    });
+
+    it('parseDatetime() should parse the date correctly', ()=>{
+        const parsedDate = parseDatetime(mockData.snippet.publishedAt);
+        expect(parsedDate).toBe("27 sept. 2014");
+    });
```
Comment (PacoMojica) [inline src/components/VideosGrid/VideoGridItem/VideoGridItem.test.js:46]: Awesome, you are testing the parser

File: test-coverage.txt [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+ PASS  src/components/Header/Searchbar/Searchbar.test.js
+ PASS  src/components/VideosGrid/VideoGridItem/VideoGridItem.test.js
+ PASS  src/components/Header/Header.test.js
+ PASS  src/components/VideosGrid/VideoGrid.test.js
+ PASS  src/components/Layout/Layout.test.js
+ PASS  src/components/App/App.test.js
+-------------------------------------|----------|----------|----------|----------|-------------------|
+File                                 |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
+-------------------------------------|----------|----------|----------|----------|-------------------|
+All files                            |    93.18 |       50 |    92.31 |    93.18 |                   |
```
Comment (PacoMojica) [inline test-coverage.txt:13]: Good code coverage, the branch percentage is a little bit low, you can improve this by adding more cases to the tests, for example if you have a something like:

```jsx
if (isOpened) {
  // code when isOpen is true
} else {
  // code when isOpen is false
}
```
the way to test this code and have a 100% coverage is by creating a case that makes `isOpen = true` and another case that makes `isOpen = false`, with this both parts of the `if` statement will be executed and the branch percentage will increase

**Review by PacoMojica — COMMENTED** [review]
Hello Alan, you have the requirements for the mini challenges 1 & 2, you tests look good but you can improve them by adding more test cases

Great work 👍

**Comment by AlanMalagon** [comment]
Thank you for the comments, Pablo, I appreciate them and I'll have them in mind for the next deliverables, especially the last comment on the branch percentage in the test coverage. About the code commented I'll remove it in the next commit then, just wanted to have the example of the changing background cause I found it useful to learn from it later.

**Pull request merged**